src/connector/ws/websocket-client.connector.ts

```diff
@@ -0,0 +1,100 @@
+import WebSocket from 'ws'
+import {
+    AnyFunction,
+    AnyFunctionWithThis,
+    OnMessageCb,
+    WsApiConfig
+} from './types'
+import {WsProviderConnector} from './websocket-provider.connector'
+
+export class WebsocketClient implements WsProviderConnector {
+    public ws!: WebSocket
+
+    public url: string
+
+    private initialized: boolean
+
+    constructor(config: WsApiConfig) {
+        const castedUrl = castUrl(config.url)
+        this.url = `${castedUrl}/v1.0/${config.network}`
```

Comment:
Only required `url` and optional `lazy init` are required here. URL casting should be on `WebSocketApi` level, since here is just implementation of provider without business logic.